JSON and GrabPast merge

karltrowbridge1 opened the pull request:
This is a combination of the work I and @wildxmxtt have been working on. A combination of JSON and a fix of the grab past function.

karltrowbridge1 committed:
Push Gitignore

Sending just the gitignore for initial commit

karltrowbridge1 committed:
Updated the gitignore

added the spotify.json so that it doesnt get leaked!

karltrowbridge1 committed:
Removing files for gitignore

karltrowbridge1 committed:
First day Introduction

karltrowbridge1 committed:
Creation of database

I have settled on using sqlite. we have created the first iteration of database and started on converting the functions in spotbot.py

karltrowbridge1 committed:
Begain to work on dupcheck method

I have began to implement new code but am keeping the old code as a point of reference for the development of newer code. the method is not done yet.

karltrowbridge1 committed:
Cleaned up and finished dupcheck and uri

This code has solely been written and not confirmed to be in working condition

karltrowbridge1 committed:
Missed save

karltrowbridge1 committed:
Small fixes to Random Song function

Should be the completion of the first attempt to implement database use. This code has only been self reviewed and not tested.

karltrowbridge1 committed:
Confirmed working database

Now running into issues with tokens but the bot has been hosted and run using the datbase. A new song was added and old songs recoreded using grab past.

Issues now with actaully adding the songs to spotify, but database is now running and confirmed functional.

karltrowbridge1 committed:
db update

karltrowbridge1 committed:
Diagnosing issues with grab_past_flag and sending off the request

karltrowbridge1 committed:
db update

karltrowbridge1 committed:
Diagnosed issue with uritxt

Didn't call a function... woops

karltrowbridge1 committed:
Diagnosing Grab Past Issues

wildxmxtt committed:
Updated setup.json

Added installed features categories

wildxmxtt committed:
Update setup.json

syntax error fix

karltrowbridge1 committed:
Fixed issues with tuples returned from SQL query

Diagnosed issues with the database returning tuples which became an issues with the line formatting.

karltrowbridge1 committed:
Commenting Code

karltrowbridge1 committed:
Recording discord user Ids in all tables now

karltrowbridge1 committed:
Altered DB architecture and updated metadata harvesting

Only using the songs table now and also recording the time stamp of messages and message ID now. Shaping up to add achievements.

karltrowbridge1 committed:
Update database.py

karltrowbridge1 committed:
Created simple leaderboard function

wildxmxtt committed:
Merge pull request #1 from wildxmxtt/database-advancement

Database advancement

karltrowbridge1 committed:
worked on nickname usage in leaderboard

karltrowbridge1 committed:
Leader Boards Added

Both All Time and This Month have been added and work as intended. Still missing the embed thumbnail

karltrowbridge1 committed:
Update spotbot.db

karltrowbridge1 committed:
Spotbot.py skeleton code

karltrowbridge1 committed:
Added milestone messages

karltrowbridge1 committed:
Crafted SQL Statement

karltrowbridge1 committed:
Fixed Check Achievement False Return

karltrowbridge1 committed:
Beginning the creation of reactChamp

karltrowbridge1 committed:
Created Reaction Champ leaderboard

karltrowbridge1 committed:
fixed miss represented number in reactions

karltrowbridge1 committed:
Added setup.json Flag for leaderboards enablement/disablement

karltrowbridge1 committed:
Database update

karltrowbridge1 committed:
Working code for playlist duration

Issues are present however with the authentication. This should most likely live in the playlist_update.py file.

karltrowbridge1 committed:
Created Duration achievements

This code has issues where the token is grabbed twice, and the time since last refresh is not handled correctly anymore resulting in a new token grabbed every single time it is called.

karltrowbridge1 committed:
Fixed Token Issues

Upon further review, the new token expiration date was not being recorder. This made it so that every single call resulted in an 'expired' token and thus renewed the token. This has been fixed.

karltrowbridge1 committed:
Check Achievements on 10th message

karltrowbridge1 committed:
Every 5th message check for achievements

math is hard. needs to be every 5th message. woops!

karltrowbridge1 committed:
Added logic to account for all achievements

EmanG7 committed:
Add command: waves

New function/command called "waves," was added to create Spotify's Wave Codes for music entered.

EmanG7 committed:
Update hlp func

Add !waves cmd help section

EmanG7 committed:
Update README.md

EmanG7 committed:
minor edits

EmanG7 committed:
Merge pull request #8 from wildxmxtt/documentation

Merge documentation to main

EmanG7 committed:
Merge pull request #6 from wildxmxtt/wavecodes

Merge wavecodes into main

EmanG7 committed:
Merge branch 'main' into reactionChamp

karltrowbridge1 committed:
Merge pull request #7 from wildxmxtt/reactionChamp

SpotBot Leaderboard and Achievements

karltrowbridge1 committed:
Database initializer made

karltrowbridge1 committed:
Miss-spelled table

Tested and the hashes of the table outputs match. Database initializes accurately.

karltrowbridge1 committed:
Removed unused database connection

wildxmxtt committed:
Merge pull request #9 from wildxmxtt/database-initialization

Database initialization

karltrowbridge1 committed:
updated json

karltrowbridge1 committed:
updated all use of the JSON

Needs the database installed

karltrowbridge1 committed:
Updated the database creation

karltrowbridge1 committed:
fixed and moved database initialization

karltrowbridge1 committed:
updated dupCheck

karltrowbridge1 committed:
Updated grab past to scan all channels

karltrowbridge1 committed:
Small fixes and flask update

Still needs work done on Grabpast as the server wide grab past functionality has trouble specifically with getting the history of discord channels using the Channel ID

karltrowbridge1 committed:
Diagnosed issue with grabpast

Missing perms and returning `none`

karltrowbridge1 committed:
history error handling for diagnostics

karltrowbridge1 committed:
adding error handling for more diagnostics

wildxmxtt committed:
Updated css, page now works on a fundemental basis
Waiting for sql db to be created. Need to format
Page a bit more to be cleaner, but all basic
functions exist.

wildxmxtt committed:
Merge branch 'webpage-installer' of https://github.com/wildxmxtt/spotbot2 into webpage-installer

karltrowbridge1 committed:
Fixed Grab past

Has some issues with the token, might need a new token on startup

karltrowbridge1 committed:
Fixed server messages, added startup token refresh

karltrowbridge1 committed:
Started DB Development

karltrowbridge1 committed:
Worked on setup info retrieval

karltrowbridge1 committed:
Database_tools.py interfaces with spotbot.py

Database_tools.py now has tools to get server specific information all from a SQL database. The infrastructure is intended to support several Discord servers being run on a single host server running spotbot.

karltrowbridge1 committed:
Local React Champ Added

karltrowbridge1 committed:
Added song names to react champ

Still need to update help and maybe further condense code

karltrowbridge1 committed:
Updated help command

karltrowbridge1 committed:
Found issue with setupinfo

karltrowbridge1 committed:
Still diagnosing issues with the setup info

karltrowbridge1 committed:
ommited server name from db

karltrowbridge1 committed:
dabases added and milestones update

karltrowbridge1 committed:
Began install script

wildxmxtt committed:
Fix discord channel logging

1. Tested fucntionality of advanced set up on flask webapp setup
2. fixed logging with channel ids
3. channel ids need to be in an array format to be read TO-DO for SQL lite integration

wildxmxtt committed:
Merge branch 'Web2DB' into pre-playlist-db

wildxmxtt committed:
working view of database

working viewing of db's however the rest of the web applicaition is down. lots of issues coming from the heavly reliance on setup.json

wildxmxtt committed:
BUGGY - Added core functionality to webapp

Core functionality of getting spotify token up with info working. Guided setup is broken currently, as well as the button to add more rows to the chat table in the db.

wildxmxtt committed:
Fixed add row on advanced setup

when a user hits the add new row button on the advance setup part of the webapp it will

wildxmxtt committed:
Fix Lots of Minor Bugs - Still Buggy

Both Continuted setup & advanced are fully functional

Bugs Fixed:
-Advanced Setup Not displaying correct items from db, after db had been updated
-All Update and submit buttons are now working
-All Non-function buttons removed
-Fixed issue on advanced setup where "ghost" records would apperar even though a user wouldnt submit into table

Known Issues:
Spotbot.py still broken due to missing spotbot.json file, need to get with @karltrowbridge1 to resolve issue.

Also need to discuss with @karltrowbridge1 about limiting discord bot tokens to just one. Or if we ever intend to have multiple discord tokens be used at once

Big update, still buggy, but getting there

karltrowbridge1 committed:
Commenting out secret DB usage

karltrowbridge1 committed:
Fixed the webapp

karltrowbridge1 committed:
JSON compatibility

JSON has been re-introduced and a database zip file has been installed for an easy installation of the spotbot database.

karltrowbridge1 committed:
Linux directory updates

karltrowbridge1 committed:
Small JSON fixes

wildxmxtt reviewed (approved):
Merge request looks good, however later on we may want to chose to be more consistent on how we are pulling from setup.json as currently we are going to ways about it for code consistency sake. Asides from that excited to see these branches combine

wildxmxtt committed:
Code needs serious debugging

Need to fix/re-implement

Playlist_update:
-Decide to implement send off2 for all instiances of sendOff?

Json Standarization
-Calling in config tools or not for playlist_update.py & spotbot.py

Co-Authored-By: Karl Trowbridge <93893956+karltrowbridge1@users.noreply.github.com>
